ide/projectapi/src/org/netbeans/spi/project/SingleMethod.java

```diff
@@ -94,14 +126,15 @@ public boolean equals(Object obj) {
             return false;
         }
         SingleMethod other = (SingleMethod) obj;
-        return other.file.equals(file) && other.methodName.equals(methodName);
+        return other.file.equals(file) && other.methodName.equals(methodName) && other.nestedClass.equals(nestedClass);
     }
 
     @Override
     public int hashCode() {
         int hash = 7;
         hash = 29 * hash + this.file.hashCode();
         hash = 29 * hash + this.methodName.hashCode();
+        hash = 29 * hash + this.nestedClass.hashCode();
```

Comment:
:warning: Objects.hashCode() to avoid NPE on nestedClass; similar for equals. Unlike the other fields, nestedclass is optional.